linkerd/app/src/lib.rs

```diff
@@ -140,6 +140,11 @@ impl Config {
         let oc_span_sink = oc_collector.span_sink();
 
         let start_proxy = Box::pin(async move {
+            let span = info_span!("outbound");
+            let _outbound = span.enter();
```

Comment:
in general, we say not to use `Span::enter` in async-await. In this case, it's alright, since the `Entered` guard isn't held across an await point, but that wasn't immediately obvious. Maybe there should be a comment about that? 

Alternatively, we could use `Span::in_scope`, which _ensures_ that the span is not entered across an await point, but that would cause some git churn (by indenting all the code inside the span)...

Reply:
fwiw, there's not actually any use of `.await` in this block -- it's just in an `async` to be lazy.

Reply:
going to hold off on these changes as it incurs more boilerplate than i'd like -- we probably need to take a more intentional pass at organizing these builders once the gateway stuff is resolved.